build/scripts/check-toolset-insertion.ps1

```diff
@@ -0,0 +1,126 @@
+Param(
+    [string]$sourcePath = $null,
+    [string]$binariesPath = $null
+)
+set-strictmode -version 2.0
+$ErrorActionPreference="Stop"
+
+[Void][Reflection.Assembly]::LoadWithPartialName('System.IO.Compression.FileSystem')
+
+try
+{
+    write-host "Verify Toolset binaries folder"
+    
+    # Files whech are deployed but do not need to be included in deployment
+    $excludedFiles = @(
+        "Toolset_DoNetUse.exe",
+        "Microsoft.Build.dll",
+        "Microsoft.Build.Framework.dll",
+        "Microsoft.Build.Tasks.Core.dll",
+        "Microsoft.Build.Utilities.Core.dll",
+        "Microsoft.DiaSymReader.Native.arm.dll"
+    )
+    
+    $deploymentPath = join-path $binariesPath "Exes\Toolset"
+    $deployedFiles = 
+        gci -re -in *.exe,*.dll $deploymentPath |
+        %{ split-path -leaf $_ } |
+        ?{ -not $excludedFiles.Contains($_) } |
+        sort-object;
+        
+    # The number 20 is somewhat arbitrary.  The goal is to guarantee we don't succeed by virtue of 
+    # verifying an empty directory.  But also don't want the number to be the exact count of assemblies
+    # in the NuSpec because it would make it more tedious to do future updates.
+    if ($deployedFiles.Count -lt 20)
+    {
+        write-host "Missing deployed files"
+        exit 1
+    }
+
+    # ----------------------------------------------------------------------------------------------------------------
+    
+    # We now perform two-way verification on both nuspec and vsix.
+    # First we check that all deployed files exist in the package, then we ensure that all package files exist in the deployment folder.
+    # Several of the files in the package are necessarily taken from a directory other than the deployment folder (for signing reasons).
+    # The reverse check ensures Toolset includes the build of all the products shipped in the package and hence the full set of facades.
+
+    $allGood = $true
+
+    write-host "Verifying contents of Micrsoft.Net.Compilers.nuspec"
+    [xml]$nuspecxml = gc (join-path $sourcePath "src\NuGet\Microsoft.Net.Compilers.nuspec")
+    $nuspecAssemblies = 
+        $nuspecxml.package.files.file | 
+        %{ $_.src } | 
+        ? { (-not $_.Contains("$")) -and (-not $_.EndsWith(".config")) -and (-not $_.EndsWith(".rsp")) } | 
```

Comment:
Behaviour was included in https://github.com/dotnet/roslyn/commit/8f22bfcc229234014c7bb388d234555b8cffb810. Will add it to this PR as well so that we have to better coverage.